src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -139,61 +141,157 @@ class BinaryCompactObject : public DomainCreator<3> {
       domain::CoordinateMap<Frame::Logical, Frame::Inertial,
                             CoordinateMaps::Wedge3D>>;
 
-  struct InnerRadiusObjectA {
-    using type = double;
-    static constexpr Options::String help = {
-        "Inner coordinate radius of Layer 1 for Object A."};
+  /// Options for one of the two objects in the binary domain
+  struct Object {
+    double inner_radius;
+    double outer_radius;
+    double x_coord;
+    std::optional<
+        std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>
+        inner_boundary_condition;
+    bool use_logarithmic_map;
+    size_t addition_to_radial_refinement_level;
+    /// Whether or not the object should be excised from the domain, leaving a
+    /// spherical hole. When this is true, `inner_boundary_condition` is
+    /// guaranteed to hold a value (though it might be a `nullptr` if we are not
+    /// working with boundary conditions).
+    bool is_excised() const noexcept;
   };
 
-  struct OuterRadiusObjectA {
-    using type = double;
+  template <bool UseBoundaryConditions, typename BoundaryConditionsBase>
```

Comment:
yep, I pushed a fixup that does these things:
- Retrieve the boundary conditions from the metavars directly in the `options` lists.
- Merge `ObjectFromOptions` into `Object` since it's not needed anymore to provide the template parameters.
- Move `Excision` out of `Object` so it's easier to construct.
- Always construct an `Object` from an `Excision`, not from the the inner boundary condition directly. For this I had to add a bunch of extra `{}` braces in the tests.